fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,822 @@
+                        kwname, "a positive number or -1"
+                    )
+                )
+
+        for kw, kwname in (
+            (self.shuffle, "shuffle"),
+            (self.skip_validation, "skip_validation"),
+            (self.warm_start, "warm_start"),
+        ):
+            if not isinstance(kw, bool):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(kwname, "a boolean")
+                )
+
+        if self.callbacks and not callable(self.callbacks):
+            if not isinstance(self.callbacks, list):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        "callbacks", "a callable or list of callables"
+                    )
+                )
+            else:
+                for cb in self.callbacks:
+                    if not callable(cb):
+                        raise ValueError(
+                            _KWARG_ERROR_MESSAGE.format(
+                                "callbacks", "a callable or list of callables"
+                            )
+                        )
+
+        # NOTE: inferring distribution type should happen before transforming
+        read_kw = (
+            lambda data, kw_or_func: _get_type(data, kw_or_func)
+            if isinstance(kw_or_func, str)
+            else kw_or_func
+        )
+
+        self.predictor_loss_ = read_kw(Y, self.predictor_loss)
+        self.adversary_loss_ = read_kw(Z, self.adversary_loss)
+        self.predictor_function_ = read_kw(Y, self.predictor_function)
+
+        kws = ["binary", "category", "continuous"]
+        for kw, kwname in (
+            (self.predictor_loss_, "predictor_loss"),
+            (self.adversary_loss_, "adversary_loss"),
+            (self.predictor_function_, "predictor_function"),
+        ):
+            # Some keywords can also be a callable instead of a str.
+            if not ((isinstance(kw, str) and kw in kws) or callable(kw)):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        kwname,
+                        (
+                            "'auto', 'binary', 'category', 'continuous', "
+                            + "or a callable",
+                        ),
+                    )
+                )
+
+        for kw, kwname in (
+            (self.y_transform, "y_transform"),
+            (self.z_transform, "z_transform"),
+        ):
+            if not (
+                issubclass(type(kw), TransformerMixin)
+                and hasattr(kw, "fit")
+                and hasattr(kw, "transform")
+            ):
+                raise ValueError(
+                    _KWARG_ERROR_MESSAGE.format(
+                        kwname,
+                        "a sklearn Transformer (subclass TransformerMixin)",
+                    )
+                )
+
+        self.y_transform_ = self.y_transform.fit(Y)
+        self.z_transform_ = self.z_transform.fit(Z)
```

Comment:
if we don't want to expose these as _public_ to the user (which I think we don't need to), I'd do them as `self._y_transformer` instead

Reply:
I agree they needn't be public. This then overrules the sklearn convention of appending a `_` to validated parameters?

Reply:
sklearn's convention is that you `attr_` is a fitted public attribute, and `_attr` is a private fitted attribute.